R/dash.R

```diff
@@ -731,7 +723,7 @@ Dash <- R6::R6Class(
                   other = other_files_map))
     },
 
-    componentify = function(x) {
+    validate_component = function(x) {
       if (is.component(x)) return(x)
       if (all(vapply(x, is.component, logical(1)))) return(x)
       stop("The layout must be a component or a collection of components", call. = FALSE)
```

Comment:
In python the error we get is:
```
dash.exceptions.NoLayoutException:
Layout must be a dash component or a function that returns a dash component.
```

As discussed on slack earlier, seems like the same holds in R, we can't have a list/collection here either. And if this is where you'd see the error if layout was given a function with an invalid return, I'd give exactly the same error message as in Python.

That has knock-on effects - so `layout = function(...) {` should turn into `layout = function(value) {`, and this whole `validate_component` might as well be 🔪 since it'll just be `if (!is.component(x)) stop(...)`